### PR TITLE
SSE 

### PR DESCRIPTION
**Pull request by antonin77**
Sent Server Event

### CODE REVIEW

File: src/main/resources/data-test.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO API_REQ(ID_REQ, VA_API_URI, VA_API_MTH, VA_API_CHR, VA_API_APP, VA_API_ENV, VA_ASR_PRL, VA_ASR_STR, VA_ASR_ENB, VA_ASR_DBG)
+VALUES(1, '/v2/simulation/turpe/simulate/convention/ACC00000001/prm/16172213990135?start=2019-06-06&end=2019-06-30&fta=BTINFCU4&fta=BTINFCU4ACC&fta=BTINFCUST&fta=BTINFLU&fta=BTINFMU4&fta=BTINFMU4ACC&fta=BTINFMUDT', 'POST', 'UTF-8', 'calcul-autoconsommation', 'pprd', TRUE, TRUE, TRUE, FALSE);
```
Comment (usfalami) [inline src/main/resources/data-test.sql]: ALERT CYBER !!

File: src/main/java/org/usf/assertapi/server/controller/MainController.java [hunk 2]
```diff
@@ -1,113 +1,101 @@
...
+				assertions.assertApi(q);
 				log.info("TEST {} OK", q);
 			}
 			catch(Throwable e) {
 				//fail
 				log.error("assertion fail", e);
 			}
 		});
-		return results;
+		sseEmitters.get(guid).complete();
```
Comment (usfalami) [inline src/main/java/org/usf/assertapi/server/controller/MainController.java]: finally ??

File: src/main/java/org/usf/assertapi/server/controller/MainController.java [hunk 3]
```diff
@@ -1,113 +1,101 @@
...
+			of(disabledIds).forEach(i-> list.stream().filter(t-> t.getId() == i).findAny().ifPresent(t-> t.getConfiguration().setEnable(false)));
+		}
 		var assertions = new ApiAssertionsFactory()
 				.comparing(config.getRefer(), config.getTarget())
 				.using(new DefaultResponseComparator())
-				.trace(a-> service.trace(ctx, a))
+				.trace(a-> {
+					traceDao.insert(ctx, a);
+					try {
+						sseEmitters.get(guid).send(SseEmitter.event().name(a.getId() + " end " + guid).data(a));
```
Comment (usfalami) [inline src/main/java/org/usf/assertapi/server/controller/MainController.java:80]: hide this in a new service

File: src/main/java/org/usf/assertapi/server/controller/MainController.java [hunk 4]
```diff
@@ -1,113 +1,101 @@
...
+					} catch (IOException e) {
+						log.error("server sent events fail", e);
+					}
+				})
```
Comment (usfalami) [inline src/main/java/org/usf/assertapi/server/controller/MainController.java]: hide this in a new service

**Comment by antonin77** [comment]
Rerun for response compare

**Pull request merged**